core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -671,10 +671,23 @@ private[deploy] class Master(
       // If the cores left is less than the coresPerExecutor,the cores left will not be allocated
       if (app.coresLeft >= coresPerExecutor) {
         // Filter out workers that don't have enough resources to launch an executor
-        val usableWorkers = workers.toArray.filter(_.state == WorkerState.ALIVE)
+        var usableWorkers = workers.toArray.filter(_.state == WorkerState.ALIVE)
           .filter(worker => worker.memoryFree >= app.desc.memoryPerExecutorMB &&
             worker.coresFree >= coresPerExecutor)
           .sortBy(_.coresFree).reverse
+
+        if (spreadOutApps) {
+          def hasContainWorkersCount(worker: WorkerInfo): Int = {
```

Comment:
Nit: this can just be `app.executors.values.count(_.worker == worker)`. It's simple enough too that you don't need a method.

Reply:
I will update it ,thanks